### PR TITLE
Added info on HPA and resource spikes

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1442319

### CODE REVIEW

**Comment by mburke5678** [comment]
@miheer @mdshuai Am I even close on this?

**Comment by mburke5678** [comment]
@DirectXMan12 In the BZ, you say the user needs to "implement a readiness hook". What is a readiness hook? We have info on readiness checks; is that the same?

**Comment by mburke5678** [comment]
@DirectXMan12 @miheer @mdshuai Bump
Please advise if I have this feature correct.

**Comment by mburke5678** [comment]
@DirectXMan12 PTAL.

**Comment by mdshuai** [comment]
LGTM, THANKS

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL. Small change for a feature added in 3.5.

File: dev_guide/pod_autoscaling.adoc [hunk 1]
```diff
@@ -73,6 +73,14 @@ of the replication controller. For deployment configurations, scaling correspond
 directly to the replica count of the deployment configuration. Note that autoscaling
 applies only to the latest deployment in the `Complete` phase.
 
+{product-title} automatically accounts for resources and prevents unnecessary autoscaling 
+during resource spikes, such as during start up. Pods in the `unready` state will 
+have `0 CPU` usage when scaling up and the autoscaler will ignore the pods when scaling down.  
+Pods without known metrics have a `0% CPU` usage when scaling up and 100% CPU when scaling down.
```
Comment (ahardin-rh) [inline dev_guide/pod_autoscaling.adoc]: For consistency, should 100% CPU be in markdown since 0% CPU is?

File: dev_guide/pod_autoscaling.adoc [hunk 2]
```diff
@@ -73,6 +73,14 @@ of the replication controller. For deployment configurations, scaling correspond
 directly to the replica count of the deployment configuration. Note that autoscaling
 applies only to the latest deployment in the `Complete` phase.
 
+{product-title} automatically accounts for resources and prevents unnecessary autoscaling 
+during resource spikes, such as during start up. Pods in the `unready` state will 
+have `0 CPU` usage when scaling up and the autoscaler will ignore the pods when scaling down.  
+Pods without known metrics have a `0% CPU` usage when scaling up and 100% CPU when scaling down.  
+This will allow for more stability during the HPA decision. To use this feature, you must configure
```
Comment (ahardin-rh) [inline dev_guide/pod_autoscaling.adoc]: s/This will allows for more /This allows more

**Comment by mburke5678** [comment]
[rev_history]
|xref:../dev_guide/pod_autoscaling.adoc#dev-guide-pod-autoscaling[Pod Autoscaling]
|Added information on how xref:../dev_guide/pod_autoscaling.adoc#hpa-autoscaling[resource startup spikes are handled to avoid autoscaling].
%

**Comment by DirectXMan12** [comment]
whoops, sorry about missing the review on this.  Looks good to me :+1:

**Comment by vikram-redhat** [comment]
@mburke5678 - this one had multiple commits. Please make sure to squash commits.

**Pull request merged**